iFixit/react-commerce
# Use new algolia hooks

closes #301 
closes #298 
closes #137 
closes #165 
closes #225 
closes #239 

This PR drops the custom Algolia implementation in favour of the new [React InstantSearch Hooks](https://www.algolia.com/doc/guides/building-search-ui/what-is-instantsearch/react-hooks/) library.
The motivation behind this change is that issue #301 requires a deep change of the custom implementation, so refactoring to the new React InstantSearch hooks library is an easier refactor and brings long term advantages including not having to maintain our custom implementation and faster deployment of new features using the readily available hooks.

Since filters component had to be refactored I took this opportunity to implement the [new design](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit-Shopify-Working-Files?node-id=3245%3A29545). 

## QA

1. Visit the Vercel PR preview (replace vercel.app with cominor.com) 
2. Verify that you can filter using encoded url (e.g. https://react-commerce-git-use-new-algolia-hooks-ifixit.cominor.com/Parts?filter%5Bworksin%5D%5B0%5D=iPhone%207%20A1660%20Verizon%2FSprint%2FChina) as suggested in #301 (note that the syntax slightly differs in that params a enclosed in an array e.g. `filter[worksin]`
3. Verify that all other issues enumerated above are fixed

qa_req 2
cr_req 2

## Notes

:warning:  that currently this branch is using an Algolia app from my personal account, because we're waiting for Algolia to enable the "Facet display" feature on iFixit account.

deploy_block 🚨
